WikiWatershed/model-my-watershed
# MapShed Compare: Water Quality Charts

## Overview

Adds GWLF-E Water Quality Charts to the Compare View, according to the mockups.

Since there are so many code changes, reviewing which is difficult without familiarity and context, I expect this to be primarily a functional review.

Connects #2910 

### Demo

![image](https://user-images.githubusercontent.com/1430060/44365258-433eb480-a497-11e8-88df-5513527bd47a.png)

### Notes

#2926 and #2928 are known issues, please ignore those if you come across them.

If you try to display the GWLF-E Water Quality Table, there will be an error in the console.

## Testing Instructions

* Check out this branch and `bundle`
* Go to [:8000/](http://localhost:8000/) and make a MapShed project. Add bunch of scenarios.
* Open the Compare View. Switch to the Water Quality tab. Ensure you see a chart for each Sediments, Total Nitrogen, Total Phosphorus, with a bar for every scenario, corresponding to the value in the dropdown.
* Select a different option from the dropdown. Ensure the charts update correctly.
* Close the Compare View. Make a modification on one or more scenarios. Open the Compare View again. Ensure you see updated values.
* Try paginating using the buttons on the top right.
